lua/powerups/client/curse.moon

```diff
@@ -67,13 +122,23 @@ timer.Create "CFC_Powerups-Curse-EmitterThink", EMITTER_INTERVAL, 0, ->
         for _ = 1, EMITTER_AMOUNT
             pos = centerPos + Vector(math.Rand(-spreadX, spreadX), math.Rand(-spreadY, spreadY), math.Rand(-spreadZ, spreadZ))
             dir = AngleRand!\Forward!
-            colorIntensity = math.Rand 0, EMMITTER_COLOR_INTENSITY
-            with part = emitter\Add EMITTER_MATERIAL, pos
-                \SetStartSize EMITTER_START_SIZE
-                \SetEndSize EMITTER_END_SIZE
+
+            weight = math.random 1, emitterTotalWeight
+            option = nil
+            for i, total in ipairs emitterWeightTotals
+                if weight <= total
+                    option = EMITTER_OPTIONS[i]
+                    break
```

Comment:
slightly neat alternative way to do this
```suggestion
            option = (->
                for i, total in ipairs emitterWeightTotals
                  return EMITTER_OPTIONS[i] if weight <= total
            )!

```

Reply:
That is neat, but I imagine defining a function on every emitter think would be less performant. Also imo it's slightly less readable, without actually compacting the code.